### PR TITLE
Migrate to 1ES template migration for official builds and mark others accordingly

### PR DESCRIPTION
**Pull request by arkalyanms**
Validation build is at https://dev.azure.com/dnceng/internal/_build/results?buildId=2410326&view=results. (SBOM publishing failure warning pending. Should not be a blocker though.)

### CODE REVIEW

File: azure-pipelines-microbuild.yml [hunk 1]
```diff
@@ -4,112 +4,117 @@ parameters:
...
         - task: CodeQL3000Finalize@0
           displayName: CodeQL Finalize
-- template: eng\common\templates\post-build\post-build.yml
-  parameters:
-    publishingInfraVersion: 3
-    enableSymbolValidation: false
-    enableSigningValidation: false
-    enableSourceLinkValidation: false
-    enableNugetValidation: false
+    - template: /eng/common/templates/post-build/post-build.yml@self
```
Comment (jjonescz) [inline azure-pipelines-microbuild.yml:114]: `/eng/common/templates/` should be replaced with `/eng/common/templates-official/`. But you will need to merge in https://github.com/dotnet/roslyn-analyzers/pull/7240 first.

File: azure-pipelines-perf.yml [hunk 2]
```diff
@@ -1,3 +1,9 @@
+parameters:
+- name: isOfficial
```
Comment (jjonescz) [inline azure-pipelines-perf.yml:2]: What is this parameter for?

File: azure-pipelines-microbuild.yml [hunk 3]
```diff
@@ -4,112 +4,117 @@ parameters:
...
-        - checkout: self
-          clean: true
-        - script: eng\common\CIBuild.cmd
-                    -configuration $(_BuildConfig)
-                    /p:OfficialBuildId=$(BUILD.BUILDNUMBER)
-                    /p:DotNetSignType=$(_SignType)
-                    /p:DotnetPublishUsingPipelines=true
-          displayName: Build and Test
-
-        # Archive NuGet packages to DevOps.
```
Comment (jjonescz) [inline azure-pipelines-microbuild.yml:63]: Consider preserving the comments.

File: azure-pipelines-perf.yml (hunk 2 above)
Comment (arkalyanms) [reply azure-pipelines-perf.yml:2]: We've been using that to mark the unofficial pipelines like PR, perf, localization, richnav, basically all ymls used in the public dnceng that are not directly producing the build for publishing to be marked accordingly - https://github.com/dotnet/vscode-csharp/pull/6972/files#diff-7915b9b726a397ae7ba6af7b9703633d21c031ebf21682f3ee7e6a4ec52837a5:~:text=parameters%3A-,isOfficial%3A%20false,-%2D%20stage%3A

Comment (Cosifne) [reply azure-pipelines-perf.yml:2]: Is this parameter used? anywhere?

**Review by Cosifne — APPROVED** [review]
A minor question about the parameter. Other looks good

File: azure-pipelines-microbuild.yml [hunk 4]
```diff
@@ -1,115 +1,116 @@
...
           displayName: CodeQL Initialize
-        - script: eng\common\cibuild.cmd
-            -configuration Release
-            -prepareMachine
-            /p:Test=false
+        - script: eng\common\cibuild.cmd -configuration Release -prepareMachine /p:Test=false
           displayName: Windows Build
         - task: CodeQL3000Finalize@0
-          displayName: CodeQL Finalize
-- template: eng\common\templates\post-build\post-build.yml
```
Comment (jjonescz) [inline azure-pipelines-microbuild.yml:109]: Why has this been removed?

File: azure-pipelines-perf.yml (hunk 2 above)
Comment (arkalyanms) [reply azure-pipelines-perf.yml:2]: removed for now, it was meant to filter out the 1ES scan

File: azure-pipelines-microbuild.yml [hunk 5]
```diff
@@ -1,115 +1,122 @@
...
+            steps:
+            - checkout: self
+              clean: true
+            - script: eng\common\CIBuild.cmd -configuration $(_BuildConfig) /p:OfficialBuildId=$(BUILD.BUILDNUMBER) /p:DotNetSignType=$(_SignType) /p:DotnetPublishUsingPipelines=true
+              displayName: Build and Test
+            templateContext:
+              outputs:
+              # Archive NuGet packages to DevOps.
+              - output: pipelineArtifact
+                path: $(Build.ArtifactStagingDirectory)/packages/$(BuildConfig)
```
Comment (jjonescz) [inline azure-pipelines-microbuild.yml:83]: Previously the path was `'artifacts\packages\$(_BuildConfig)'` now this seems to resolve to `D:\a\_work\1\a\packages\Release` which looks different.

**Pull request merged**